pallets/stake/src/lib.rs

```diff
@@ -615,6 +620,22 @@ decl_module! {
 		/// Minimum stake for any registered on-chain account to become a nominator
 		const MinNominatorStk: BalanceOf<T> = T::MinNominatorStk::get();
 
+		#[weight = 0]
+		fn set_blocks_per_round(
+			origin,
+			blocks_per_round: u32,
+		) -> DispatchResult {
+			frame_system::ensure_root(origin)?;
+			ensure!(
+				blocks_per_round >= T::MinBlocksPerRound::get(),
+				Error::<T>::BlocksPerRoundBelowMin
+			);
+			let mut current = <BlocksPerRound>::get();
+			current.set(blocks_per_round);
+			<BlocksPerRound>::put(current);
+			Self::deposit_event(RawEvent::BlocksPerRoundSet(blocks_per_round));
```

Comment:
should emit the block when the round will change such that the interval thereafter is the passed in `blocks_per_round`

```rust
let now = system::block_number();
let mut t = now / old;
let current_round_would_end = (t + 1) * old;
let mut tt = current_round_would_end / new;
let current_round_will_end = (tt + 1) * new;
```